### ISSUE TITLE
Remove the mature models from the API

### ISSUE THREAD

**Issue by AetherUnbound** [WordPress/openverse#2630]
## Description

Remove the old mature models (`MatureImage`, `MatureAudio`, `NsfwReport`, `NsfwReportAudio`) from the API, which will necessitate a migration. Instructions for how to perform this change are present in the implementation plan associated with this work: https://docs.openverse.org/projects/proposals/trust_and_safety/content_report_moderation/20230517-implementation_plan_mature_copy_change.html#code (step 9).

A deployment of the API will be necessary for this change to be considered complete, as a migration is involved.

## Additional context

Part of #383, blocked by #2629

**Comment by AetherUnbound** [comment]
Closing in favor of #3723 based on #3568.
